CONTRIBUTING.md

```diff
@@ -1,183 +1,71 @@
-      attr1 (str): Description of `attr1`.
-      attr2 (:obj:`int`, optional): Description of `attr2`.
-
-  """
-
-  def __init__(self, param1, param2, param3):
-      """Example of docstring on the __init__ method.
+### Reporting bugs
 
-      The __init__ method may be documented in either the class level
-      docstring, or as a docstring on the __init__ method itself.
+Before submitting a new bug check if the bug is not already reported in the [bug issues](https://github.com/ispyb/py-ispyb/issues?q=is%3Aopen+is%3Aissue+label%3Abug).
+If the corresponding issue does not exist then:
 
-      Either form is acceptable, but the two should not be mixed. Choose one
-      convention to document the __init__ method and be consistent with it.
+-   [Open a new issue](https://github.com/ispyb/py-ispyb/issues/new) with a short description in the title.
+-   In the description describe the bug:
+    -   Conditions when the bug appears;
+    -   How it can be reproduced;
+    -   Possible cause of the bug and source code where it occurs;
+    -   If possible add an error log.
+-   Assign the `bug` label to the issue.
 
-      Note:
-          Do not include the `self` parameter in the ``Args`` section.
+### Submitting code to the repository
 
-      Args:
-          param1 (str): Description of `param1`.
-          param2 (:obj:`int`, optional): Description of `param2`. Multiple
-              lines are supported.
-          param3 (list(str)): Description of `param3`.
+To submit code to the repository, please follow these steps:
 
-      """
-      self.attr1 = param1
-      self.attr2 = param2
-      self.attr3 = param3  #: Doc comment *inline* with attribute
+1. All code contribution has to be done from an issue. If there is no existing issue for the submission you wish to make, start by [creating a new one](https://github.com/ispyb/py-ispyb/issues/new), and describe what should be done for it to be considered as fulfilled;
 
-      #: list(str): Doc comment *before* attribute, with type specified
-      self.attr4 = ['attr4']
+2. Once you have identified your contribution issue, [assign it to yourself](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) so that everyone can keep track of what is in progress;
 
-      self.attr5 = None
-      """str: Docstring *after* attribute, with type specified."""
+3. Create a branch for the contribution. You should create a branch from the GitHub issue page, by clicking the `Create a branch` button under the `Development` section on the right;
 
-```
+4. Start implementing your changes and [commit](https://github.com/git-guides/git-commit) them;
 
-An example how to describe a function:
+5. Once the changes are mature enough to be discussed, create a `draft` [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) in the repository;
```

Comment:
One suspects that in simpler cases (bug fixes?) people might want to short-circuit this a bit. How strict do we want to be on this?